### PR TITLE
Only save computed diagnostic

### PR DESCRIPTION
**Pull request by navidcy**
This PR modifies `savediagnostic` slightly so it only saves the diagnostic that were computed, i.e., `1 : diag.i`.

### CODE REVIEW

File: src/output.jl [hunk 1]
```diff
@@ -147,13 +147,14 @@ saveproblem(out::Output) = saveproblem(out.prob, out.path)
 """
     savediagnostic(diag, diagname, filename)
 
-Save diagnostics in `diag` to `filename`, labeled by `diagname`.
+Save diagnostic in `diag` to `filename` under name `diagname`. Only the computed diagnostic
```
Comment (jbisits) [inline src/output.jl:150]: This is clear but could maybe change to 

Save the diagnostic `diag`...

after is good.

**Review by jbisits — APPROVED** [review]
I added one minor phrasing comment, but if you like the way it is at the moment I think that is fine! Otherwise all looks good!

File: src/output.jl (hunk 1 above)
Comment (navidcy) [reply src/output.jl]: I changed everything to `diagnostic` instead of `diag`.

File: src/output.jl [hunk 2]
```diff
@@ -145,15 +145,16 @@ end
 saveproblem(out::Output) = saveproblem(out.prob, out.path)
 
 """
-    savediagnostic(diag, diagname, filename)
+    savediagnostic(diagnostic, diagname, filename)
 
-Save diagnostics in `diag` to `filename`, labeled by `diagname`.
+Save `diagnostic` to `filename` under name `diagname`. Only the computed diagnostic
+is saved, that is, everything up to diagnostic's iteration `diagnostics.i`.
```
Comment (jbisits) [inline src/output.jl:151]: `diagnostics.i` should be `diagnostic.i`?

**Review by jbisits — CHANGES_REQUESTED** [review]
I think there is one small modification to the `savediagnostic()` docstring that needs to be made. Apart from that I think this is much clearer and you are right to merge!

File: src/output.jl (hunk 2 above)
Comment (navidcy) [reply src/output.jl:151]: Yes. Good catch.

**Review by jbisits — APPROVED** [review]

**Pull request merged**